examples/medplum-provider/src/pages/messages/MessagesPage.tsx

```diff
@@ -0,0 +1,134 @@
+import { ScrollArea, Text, Loader, Grid, Flex, Paper, Group } from '@mantine/core';
+import { Communication, Patient, Reference } from '@medplum/fhirtypes';
+import { useMedplum, BaseChat, PatientSummary, useMedplumProfile } from '@medplum/react';
+import { JSX, useEffect, useState, useMemo, useCallback } from 'react';
+import { createReference, getReferenceString } from '@medplum/core';
+import { ChatList } from '../../components/messages/ChatList';
+
+/**
+ * Messages page that matches the Home page layout but without the patient list.
+ * @returns A React component that displays the messages page.
+ */
+export function MessagesPage(): JSX.Element {
+  const medplum = useMedplum();
+  const [loading, setLoading] = useState(false);
+  const [selectedPatient, setSelectedPatient] = useState<Reference<Patient> | undefined>(undefined);
+  const [inProgressCommunications, setInProgressCommunications] = useState<Communication[]>([]);
+  const [threadMessages, setThreadMessages] = useState<Communication[]>([]);
+  const profile = useMedplumProfile();
+  const profileRef = useMemo(() => (profile ? createReference(profile) : undefined), [profile]);
+
+  useEffect(() => {
+    async function fetchAllCommunications(): Promise<void> {
+      if (selectedPatient) {
+        return;
+      }
+
+      const searchParams = new URLSearchParams();
+      searchParams.append('_sort', '-sent');
+      searchParams.append('received:missing', 'true');
+      const searchResult = await medplum.searchResources('Communication', searchParams, { cache: 'no-cache' });
+
+      const threads: Communication[] = [];
+      const threadMap: Set<string> = new Set<string>();
+      searchResult.forEach((c) => {
+        if (c.subject?.reference && !threadMap.has(c.subject.reference)) {
+          threadMap.add(c.subject.reference);
+          threads.push(c);
+        }
+      });
+
+      setInProgressCommunications(threads);
+
+      if (threads.length > 0) {
+        setSelectedPatient(threads[0].subject as Reference<Patient>);
+      }
+    }
+    fetchAllCommunications().catch(() => setLoading(false));
+  }, [medplum, selectedPatient]);
+
+  const sendMessage = useCallback(
+    (message: string) => {
+      if (!selectedPatient) {
+        return;
+      }
+
+      if (!profileRef) {
+        return;
+      }
+
+      medplum
+        .createResource<Communication>({
+          resourceType: 'Communication',
+          status: 'in-progress',
+          sender: profileRef,
+          subject: selectedPatient,
+          recipient: [selectedPatient],
+          sent: new Date().toISOString(),
+          payload: [{ contentString: message }],
+        })
+        .catch(console.error);
```

Comment:
Similar question here; i'm not familiar with how polished error handling is throughout the provider app, but should this show a toast?